### PR TITLE
tiproxy: add doc for load balance

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

tiproxy: add doc for load balance

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17485
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiproxy/tiproxy-overview.md [hunk 1]
```diff
@@ -44,12 +48,14 @@ TiProxy is suitable for the following scenarios:
...
 - Frequent scaling in and scaling out: The workload of an application might change periodically. To save costs, you can deploy TiDB on the cloud and automatically scale in and scale out TiDB servers according to the workload. However, scaling in might cause the client to disconnect, and scaling out might result in unbalanced load. TiProxy can keep the client connection and achieve load balancing.
+- CPU load imbalance: When background tasks consume a significant amount of CPU resources or workloads across connections vary significantly, leading to an imbalanced CPU load, TiProxy can migrate connections based on CPU usage to achieve load balancing. For more details, see [CPU-based load balancing](/tiproxy/tiproxy-load-balance.md#cpu-based-load-balancing).
+- TiDB server OOM: When a runaway query causes a TiDB server  to run out of memory, TiProxy can proactively detect the OOM risk and migrate other healthy connections to different TiDB servers, thus ensuring continuous client connectivity. For more details, see [Memory-based load balancing](/tiproxy/tiproxy-load-balance.md#memory-based-load-balancing).
```
Comment (djshow832) [inline tiproxy/tiproxy-overview.md:52]:
```suggestion
- TiDB server OOM: When a runaway query causes a TiDB server to run out of memory, TiProxy can proactively detect the OOM risk and migrate other healthy connections to different TiDB servers, thus ensuring continuous client connectivity. For more details, see [Memory-based load balancing](/tiproxy/tiproxy-load-balance.md#memory-based-load-balancing).
```

**Review by djshow832 — APPROVED** [review]

File: tiproxy/tiproxy-load-balance.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+By default, this policy has a lower priority than health-based, memory-based, and CPU-based load balancing policies. You can increase its priority by setting [`policy`](/tiproxy/tiproxy-configuration.md#policy) to `location`. To maintain availability and performance, it is recommended to ensure that at least three TiDB servers are in the same location.
+
+TiProxy determines the locations of itself and TiDB servers based on the `zone` label. You need to set the following configuration items:
+
+- In the [`labels`](/tidb-configuration-file.md#labels) configuration item of the TiDB server, set `zone` to the current availability zone. For configuration details, see [Configure labels for TiDB](/schedule-replicas-by-topology-labels.md#optional-configure-labels-for-tidb).
+- In the [`labels`](/tiproxy/tiproxy-configuration.md#labels) configuration item of TiProxy, set `zone` to the current availability zone.
+
+For clusters deployed using TiDB Operator, see [High availability of data
+](https://docs.pingcap.com/tidb-in-kubernetes/stable/configure-a-tidb-cluster#high-availability-of-data).
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-load-balance.md:75]:
```suggestion
For clusters deployed using TiDB Operator, see [High availability of data](https://docs.pingcap.com/tidb-in-kubernetes/stable/configure-a-tidb-cluster#high-availability-of-data).
```

File: tiproxy/tiproxy-load-balance.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+TiProxy migrates connections from a TiDB server with more connections to those with fewer connections. This policy is not configurable and has the lowest priority.
+
+Typically, TiProxy identifies the load on TiDB servers based on CPU usage. This policy usually takes effect in the following scenarios:
+
+- When the TiDB cluster is just starting up, the CPU usage of all TiDB servers is close to 0. In this case, this policy prevents imbalanced load during startup.
+- When the [CPU-based load balancing](#cpu-based-load-balancing) is not enabled, this policy ensures load balancing.
+
+## Configure load balancing policies
+
+TiProxy supports configuring the combination and priority of the preceding load balancing policies through the configuration item [`policy`](/tiproxy/tiproxy-configuration.md#policy).
```
Comment (qiancai) [inline tiproxy/tiproxy-load-balance.md:124]:
```suggestion
TiProxy lets you configure the combination and priority of load balancing policies through the [`policy`](/tiproxy/tiproxy-configuration.md#policy) configuration item.
```

File: tiproxy/tiproxy-load-balance.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+- `resource`: the resource priority policy performs load balancing based on the following priority order: status, health, memory, CPU, location, and connection count.
+- `location`: the location priority policy performs load balancing based on the following priority order: status, location, health, memory, CPU, and connection count.
+- `connection`: the minimum connection count policy performs load balancing based on the following priority order: status and connection count.
```
Comment (qiancai) [inline tiproxy/tiproxy-load-balance.md:128]:
```suggestion
- `connection`: the minimum connection count priority policy performs load balancing based on the following priority order: status and connection count.
```

File: tiproxy/tiproxy-overview.md [hunk 5]
```diff
@@ -30,6 +30,10 @@ Connection migration usually occurs in the following scenarios:
 - When a TiDB server performs scaling in, rolling upgrade, or rolling restart, TiProxy can migrate connections from the TiDB server that is about to be offline to other TiDB servers to keep the client connection alive.
 - When a TiDB server performs scaling out, TiProxy can migrate existing connections to the new TiDB server to achieve real-time load balancing without resetting the client connection pool.
 
+### Failover
+
+When a TiDB server is at risk of running out of memory (OOM) or fails to connect to PD or TiKV, TiProxy detects the issue and migrates the connections to other TiDB servers, thus ensuring continuous client connectivity.
```
Comment (qiancai) [inline tiproxy/tiproxy-overview.md:35]:
```suggestion
When a TiDB server is at risk of running out of memory (OOM) or fails to connect to PD or TiKV, TiProxy detects the issue automatically and migrates the connections to other TiDB servers, thus ensuring continuous client connectivity.
```

File: tiproxy/tiproxy-overview.md [hunk 6]
```diff
@@ -44,12 +48,14 @@ TiProxy is suitable for the following scenarios:
...
 
 TiProxy is not suitable for the following scenarios:
 
 - Sensitive to performance: The performance of TiProxy is lower than that of HAProxy and other load balancers, so using TiProxy will reduce the QPS. For details, refer to [TiProxy Performance Test Report](/tiproxy/tiproxy-performance-test.md).
 - Sensitive to cost: If the TiDB cluster uses hardware load balancers, virtual IP, or the load balancer provided by Kubernetes, adding TiProxy will increase the cost. In addition, if you deploy the TiDB cluster across availability zones on the cloud, adding TiProxy will also increase the traffic cost across availability zones.
-- TiDB server failover: TiProxy can keep the client connection only when the TiDB server is offline or restarted as planned. If the TiDB server is offline unexpectedly, the connection is still broken.
+- Unexpected TiDB server downtime failover: TiProxy can keep the client connection only when the TiDB server is offline or restarted as planned. If the TiDB server is offline unexpectedly, the connection is still broken.
```
Comment (qiancai) [inline tiproxy/tiproxy-overview.md:58]:
```suggestion
- Failover for unexpected TiDB server downtime: TiProxy can keep the client connection only when the TiDB server is offline or restarted as planned. If the TiDB server is offline unexpectedly, the connection is still broken.
```

File: tiproxy/troubleshoot-tiproxy.md [hunk 7]
```diff
@@ -30,9 +30,13 @@ You can troubleshoot the issue by following these steps:
 
 ## Unbalanced CPU usage on TiDB server
 
-Check if the number of connections on TiDB server is balanced. If not, troubleshoot by following the [TiProxy does not migrate connections](#tiproxy-does-not-migrate-connections) section.
+You can troubleshoot the issue by following these steps:
 
-If the number of connections is balanced, it might be that some connections occupy a high CPU usage while other connections are relatively idle. TiProxy balances connections based on the number of connections on the TiDB server, not the actual load.
+1. Check if there is a significant difference in CPU usage among TiDB servers. TiProxy does not guarantee identical CPU usage across TiDB servers. It only performs [load balancing](/tiproxy/tiproxy-load-balance.md) when the CPU usage difference is large enough to impact query latency.
```
Comment (qiancai) [inline tiproxy/troubleshoot-tiproxy.md:35]:
```suggestion
1. Check if there is a significant difference in CPU usage among TiDB servers. TiProxy does not guarantee identical CPU usage across TiDB servers. It only performs [load balancing](/tiproxy/tiproxy-load-balance.md) when the CPU usage difference is large enough to affect query latency.
```

File: tiproxy/troubleshoot-tiproxy.md [hunk 8]
```diff
@@ -30,9 +30,13 @@ You can troubleshoot the issue by following these steps:
...
+2. If the connection count of a TiDB server gradually drops to zero, it might be affected by other load balancing policies. You can check the [`Session Migration Reasons`](/tiproxy/tiproxy-grafana.md#balance) metric in Grafana to see if there are migrations based on other policies.
+3. Check if the TiProxy configuration item [`policy`](/tiproxy/tiproxy-configuration.md#policy) is set to `location`. If location-based prioritization is enabled, TiProxy does not balance CPU usage across different locations.
+4. Check the version of TiProxy. Only v1.1.0 and later versions support CPU-based load balancing. Earlier versions use a minimum connection count policy for load balancing.
```
Comment (qiancai) [inline tiproxy/troubleshoot-tiproxy.md:38]:
```suggestion
4. Check the version of TiProxy. Only v1.1.0 and later versions support CPU-based load balancing. Earlier versions use a load balancing policy based on minimum connection count.
```

File: tiproxy/tiproxy-load-balance.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+- When the TiDB cluster is just starting up, the CPU usage of all TiDB servers is close to 0. In this case, this policy prevents imbalanced load during startup.
+- When the [CPU-based load balancing](#cpu-based-load-balancing) is not enabled, this policy ensures load balancing.
+
+## Configure load balancing policies
+
+TiProxy lets you configure the combination and priority of load balancing policies through the [`policy`](/tiproxy/tiproxy-configuration.md#policy) configuration item.
+
+- `resource`: the resource priority policy performs load balancing based on the following priority order: status, health, memory, CPU, location, and connection count.
+- `location`: the location priority policy performs load balancing based on the following priority order: status, location, health, memory, CPU, and connection count.
+- `connection`: the minimum connection count priority policy performs load balancing based on the following priority order: status and connection count.
```
Comment (qiancai) [inline tiproxy/tiproxy-load-balance.md:129]:
```suggestion
- `connection`: the minimum connection count policy performs load balancing based on the following priority order: status and connection count.
```

Comment (qiancai) [reply tiproxy/tiproxy-load-balance.md:129]: see https://github.com/pingcap/docs-cn/pull/17485#discussion_r1639634511

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**